File: in_Java/1221 Split a String in Balanced Strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int res = 0;
        int cur = 0;
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == 'L') cur--;
            else cur++;
            if(cur == 0){
                res++;
            }
        }
        return res;
    }
}

File: in_Python/1221 Split a String in Balanced Strings.py
class Solution(object):
    def balancedStringSplit(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        cur = 0
        for i in range(len(s)):
            if s[i] == 'R':
                cur -= 1
            else:
                cur += 1
            if cur == 0:
                res += 1
        return res
        